src/courseware/course/Course.test.jsx

```diff
@@ -378,7 +380,50 @@ describe('Course', () => {
     // We are in the middle of the sequence, so no
     expect(previousSequenceHandler).not.toHaveBeenCalled();
     expect(nextSequenceHandler).not.toHaveBeenCalled();
-    expect(unitNavigationHandler).toHaveBeenCalledTimes(4);
+    expect(unitNavigationHandler).toHaveBeenCalledTimes(2);
+  });
+
+  it('check tab navigations', async () => {
+    const courseMetadata = Factory.build('courseMetadata');
+    const unitBlocks = Array.from({ length: 3 }).map(() => Factory.build(
+      'block',
+      { type: 'vertical' },
+      { courseId: courseMetadata.id },
+    ));
+    const testStore = await initializeTestStore({ courseMetadata, unitBlocks }, false);
+    const { courseware, models } = testStore.getState();
+    const { courseId, sequenceId } = courseware;
+    const testData = {
+      ...mockData,
+      courseId,
+      sequenceId,
+      unitId: Object.values(models.units)[1].id, // Corner cases are already covered in `Sequence` tests.
+    };
+    render(<Course {...testData} />, { store: testStore });
+
+    loadUnit();
+
+    const breadcrumb = screen.getByRole('navigation', { name: 'breadcrumb' });
+    const listItems = within(breadcrumb).getAllByRole('listitem');
+    const links = listItems.map((li) => within(li).getByRole('link'));
+
+    links[0].focus();
+    expect(links[0]).toHaveFocus();
+
+    links.slice(1).forEach((link) => {
+      userEvent.tab();
+      expect(link).toHaveFocus();
+    });
+    expect(links[links.length - 1]).toHaveFocus();
+
+    userEvent.tab();
+    const notificationButton = screen.getByRole('button', { name: 'Show notification tray' });
```

Comment:
[nit]: Please try to use JS translations

```suggestion
    const notificationButton = screen.getByRole('button', { name: {int.TRANSLATION_MSG} });
```

Reply:
need to import messages from other component. Is this okay?

Reply:
on previous tests in this file used this variant
also, messages don't has this text

Reply:
This text has been [changed](https://github.com/raccoongang/frontend-app-learning/blob/aspen-release/palm.4/src/courseware/course/messages.js#L11), it is now called `Notifications tray`

Reply:
Fixed